src/components/KSlideout/KSlideout.vue

```diff
@@ -66,10 +123,51 @@ const props = defineProps({
     type: Number,
     default: 0,
   },
+  hasOnlyTitle: {
+    type: Boolean,
+    default: false,
+  },
+  title: {
+    type: String,
+    default: '',
+  },
+  badgeValue: {
+    type: String,
+    default: '',
+  },
+  badgeAppearance: {
+    type: String as PropType<BadgeAppearance>,
+    required: false,
+    validator: (value: string): boolean => {
+      return Object.keys({ ...BadgeAppearances }).includes(value)
+    },
+    default: 'warning',
+  },
+  badgeShape: {
+    type: String as PropType<BadgeShape>,
+    required: false,
+    validator: (value: string): boolean => {
+      return Object.keys({ ...BadgeShapes }).includes(value)
+    },
+    default: 'rectangular',
+  },
+  iconType: {
+    type: String,
+    default: 'copy',
+  },
+  iconColor: {
+    type: String,
+    default: 'var(--blue-500)',
+  },
+  iconSize: {
+    type: String,
+    default: '16',
+  },
```

Comment:
This seems a little over-complicated. 

Rather than having all of these extra props and logic with badges and icons and what not, why not just provide a "header" or "title" `<slot>` where the consuming app can pass in whatever arbitrary html they want into the "header" or "title" section (in our case the kbadge + text + clipboard copy icon). Keeps this component more simple, and offloads any "custom" header logic to the consuming app. 

Keep the "title" prop, and by default the "title" <slot> will just display that title with default truncating logic

Reply:
I agree here. 

Alternatively, the title could be a prop string, and then provide slots for the badge and a separate slot for the icon? 

It doesn’t make sense to add a prop for the icon since we are soon moving to a model where all icons are slotted